google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStoragePath.java

```diff
@@ -103,15 +103,24 @@ boolean seemsLikeADirectoryAndUsePseudoDirectories(Storage storage) {
     if (!prefix.endsWith("/")) {
       prefix += "/";
     }
-    Page<Blob> list =
-        storage.list(
-            this.bucket(),
-            Storage.BlobListOption.prefix(prefix),
-            // we only look at the first result, so no need for a bigger page.
-            Storage.BlobListOption.pageSize(1),
-            fileSystem.provider().getProject() == null
-                ? null
-                : Storage.BlobListOption.userProject(fileSystem.provider().getProject()));
+    String userProject = fileSystem.config().userProject();
```

Comment:
Am I understanding correctly, the bug was reading `projectId` from the `ServiceOptions` instead of from the `CloudStorageConfig`? If so, can we update the commit comment to explain this fact?

Reply:
Good call, that's much clearer.